codec/test_codec.go

```diff
@@ -414,20 +425,75 @@ func TestPointerToStruct(codec GeneralCodec, t testing.TB) {
 	require.Equal(myPtr, myPtrUnmarshaled)
 }
 
+func TestStructWithPtr(codec GeneralCodec, t testing.TB) {
```

Comment:
can we merge these two tests (`TestStructWithPtrNil` and `TestStructWithPtr`) into one?